### PR TITLE
feat: Add chaining support in TokenFileWebIdentityCredentials

### PR DESCRIPTION
**Pull request by trivikr**
##### Checklist

- [x] `npm run test` passes
- [x] changelog is added, `npm run add-change`

### CODE REVIEW

File: test/credentials.spec.js [hunk 1]
```diff
@@ -1474,6 +1490,48 @@
...
+          const sourceCredentials = {
+            AccessKeyId: 'AccessKeyIdSource',
+            SecretAccessKey: 'SecretAccessKeySource'
+          };
+          var assumeRoleSpy1 = helpers.spyOn(credentials.service, 'assumeRoleWithWebIdentity').andCallFake(function(params, cb) {
+            return cb(null, {
+              Credentials: sourceCredentials
+            });
+          });
+          var assumeRoleSpy2 = helpers.spyOn(credentials.service, 'assumeRole').andCallFake(function(params, cb) {
```
Comment (AllanFly120) [inline test/credentials.spec.js]: nitpick: I'm not a big fan of the naming here. How about `assymeRoleSpy` and `assumeRoleWithWebIdentitySpy`?

File: test/credentials.spec.js [hunk 2]
```diff
@@ -1474,6 +1490,48 @@
         });
       });
 
+      it('supports chaining in absense of web_identity_token_file and has source_profile ', function(done) {
+        delete process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+        helpers.spyOn(AWS.util, 'getProfilesFromSharedConfig').andReturn(
+          {
+            'source': {
+              'web_identity_token_file': 'cfgTokenFileSource',
+              'role_arn': 'cfgRoleArnSource'
```
Comment (AllanFly120) [inline test/credentials.spec.js]: Can you add a test of ignoring the `source_profile` in the profile with `web_identity_token_file`? You can just add this line in this test case.

File: test/credentials.spec.js (hunk 1 above)
Comment (trivikr) [reply test/credentials.spec.js]: Renamed in https://github.com/aws/aws-sdk-js/pull/2837/commits/5b24b41216f8d2ee939d8b81e8a20cf21a2b4432

File: test/credentials.spec.js (hunk 2 above)
Comment (trivikr) [reply test/credentials.spec.js]: Added test in https://github.com/aws/aws-sdk-js/pull/2837/commits/5b24b41216f8d2ee939d8b81e8a20cf21a2b4432

File: lib/credentials/token_file_web_identity_credentials.js [hunk 3]
```diff
@@ -84,11 +84,24 @@ AWS.TokenFileWebIdentityCredentials = AWS.util.inherit(AWS.Credentials, {
       );
     }
 
-    return {
+    var paramsArray = [];
+
+    while (!profile['web_identity_token_file'] && profile['source_profile']) {
+      paramsArray.unshift({
+        roleArn: profile['role_arn'],
+        roleSessionName: profile['role_session_name']
```
Comment (AllanFly120) [inline lib/credentials/token_file_web_identity_credentials.js]: Is there any other parameters you need to send with `assumerole` call? From what I can find [SharedIniCredentials](https://github.com/aws/aws-sdk-js/blob/88b635be22df3207ab7971d9b3192b5523a0a8b8/lib/credentials/shared_ini_file_credentials.js#L229-L231) can also take `ExternalId`.

Comment (trivikr) [reply lib/credentials/token_file_web_identity_credentials.js]: Nope, only RoleArn and RoleSessionName are necessary for this use case https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/STS.html#assumeRole-property

Comment (trivikr) [reply lib/credentials/token_file_web_identity_credentials.js]: Confirmed internally that only RoleArn and RoleSessionName are necessary for assumeRole call

**Pull request merged**